archive/common/src/main/scala/uk/ac/wellcome/platform/archive/common/models/DisplayIngest.scala

```diff
@@ -8,21 +10,33 @@ import uk.ac.wellcome.platform.archive.common.progress.models.{
   Resource
 }
 
-case class DisplayIngest(id: String,
-                         uploadUrl: String,
-                         callback: Option[DisplayCallback],
-                         ingestType: DisplayIngestType,
-                         space: DisplayStorageSpace,
-                         status: DisplayIngestStatus,
-                         resources: Seq[DisplayIngestResource],
-                         events: Seq[DisplayProgressEvent] = Seq.empty,
-                         createdDate: String,
-                         lastModifiedDate: String,
-                         @JsonKey("type")
-                         ontologyType: String = "Ingest")
+sealed trait DisplayIngest
+
+case class RequestDisplayIngest(uploadUrl: String,
+                                callback: Option[DisplayCallback],
+                                ingestType: DisplayIngestType,
+                                space: DisplayStorageSpace,
+                                @JsonKey("type")
+                                ontologyType: String = "Ingest")
+    extends DisplayIngest
+
+case class ResponseDisplayIngest(id: UUID,
+                                 uploadUrl: String,
+                                 callback: Option[DisplayCallback],
+                                 ingestType: DisplayIngestType,
+                                 space: DisplayStorageSpace,
+                                 status: DisplayIngestStatus,
+                                 resources: Seq[DisplayIngestResource] =
+                                   Seq.empty,
+                                 events: Seq[DisplayProgressEvent] = Seq.empty,
+                                 createdDate: String,
+                                 lastModifiedDate: String,
+                                 @JsonKey("type")
+                                 ontologyType: String = "Ingest")
+    extends DisplayIngest
 
 case class DisplayCallback(uri: String,
-                           status: String,
+                           status: Option[String],
```

Comment:
When can this be `None`?

Reply:
When it's supplied as part of a POST. I should have adopted the same approach as DiplayIngest to be honest and split this in two classes